src/pal/src/include/pal/stackstring.hpp

```diff
@@ -44,34 +52,42 @@ class StackString
             return;
         }
 
+        if(preserve)
```

Comment:
It seems that using PAL_realloc instead of PAL_malloc + CopyMemory when the preserve is true would be better. In some cases, the allocator may be able extend the allocated region "in place" without copying anything.